### PR TITLE
Migrate `ListTile` unselected icon color to Material 3

### LINKED ISSUE #99400: Migrate `ListTile` disabled icon color to Material 3
## Use case
Trailing IconButtons in ListTiles.

## Proposal
`ListTileThemeData` doesn't inherit its `iconColor` from `iconTheme`. Took me a few minutes to figure out. It would make sense that if no `iconColor` is provided to `ThemeData.listTileTheme`, the `iconColor` should default to the one in `iconTheme` instead of the default Materiel grey color.

Would you consider changing this behaviour ?

Short reproduction code :
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(
        iconTheme: IconThemeData(color: Colors.red),
      ),
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Column(
          children: const [
            ListTile(
              leading: Icon(Icons.add_rounded),
            ),
            Icon(Icons.add_rounded),
          ]
        ),
      ),
    );
  }
}
```

### PR DESCRIPTION
**Pull request by TahaTesser**
Updates `ListTile` default unselected icon color to Material 3.

Fixes https://github.com/flutter/flutter/issues/99400

Part of: https://github.com/flutter/flutter/issues/91605

Disabled icon specs from https://m3.material.io/components/buttons/specs

Screenshot 2022-04-18 at 15 02 39

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by TahaTesser** [comment]
Just resolved conflicts

File: packages/flutter/lib/src/material/list_tile.dart [hunk 1]
```diff
@@ -588,7 +588,10 @@ class ListTile extends StatelessWidget {
       return selectedColor ?? tileTheme.selectedColor ?? theme.listTileTheme.selectedColor ?? theme.colorScheme.primary;
     }
 
-    final Color? color = iconColor ?? tileTheme.iconColor ?? theme.listTileTheme.iconColor;
+    final Color? color = iconColor
+      ?? tileTheme.iconColor
+      ?? theme.listTileTheme.iconColor
+      ?? (theme.useMaterial3 ? theme.colorScheme.onSurface.withOpacity(0.38) : null);
```
Comment (darrenaustin) [inline packages/flutter/lib/src/material/list_tile.dart:594]: Can we add a comment describing where the `onSurface.withOpacity` came from in the spec as this is not going through the usual token template system.

**Review by darrenaustin — APPROVED** [review]
LGTM. Sorry for the lag on the response.

File: packages/flutter/lib/src/material/list_tile.dart (hunk 1 above)
Comment (TahaTesser) [reply packages/flutter/lib/src/material/list_tile.dart:602]: Thanks! I added a comment for the color, similar to the `icon_button.dart` docs.
https://github.com/flutter/flutter/blob/d8fdb83555522dbf25659e6509ccc3551bcc0193/packages/flutter/lib/src/material/icon_button.dart#L799

Comment (guidezpl) [reply packages/flutter/lib/src/material/list_tile.dart:602]: The disabled case is handled above on line 589. So this seems to be returning the incorrect color for enabled, non-selected list tile icons.

Comment (TahaTesser) [reply packages/flutter/lib/src/material/list_tile.dart:602]: `line 589` disabled color is for disabled `list_tile`, this wasn't the fix.

The wording was wrong in the issue and PR, apolgies.

By disabled, it means unselected. This is the author's code, `ListTile` is not disabled.
```dart
     home: Scaffold(
        body: Column(
          children: const [
            ListTile(
              leading: Icon(Icons.add_rounded),
            ),
            Icon(Icons.add_rounded),
          ]
        ),
      ),
```

> The disabled case is handled above on line 589. So this seems to be returning the incorrect color for enabled, non-selected list tile icons.

In this case, I agree. At the time there were no specs for the list item but we have now and I can see the difference you're talking about.

I did a quick comparison here with Flutter vs Compose.

| Flutter | Compose |
| --------------- | --------------- |
|  | 


`ListTile` needs a complete migration, just for unselected color.


Filed https://github.com/flutter/flutter/issues/114006, please take a look.

Comment (guidezpl) [reply packages/flutter/lib/src/material/list_tile.dart:602]: Thanks, I think we'll be able to workaround this internally.

**Pull request merged**